src/logger_dispatch_overrides.jl

```diff
@@ -142,7 +167,7 @@ function preprocess(name, val::TBImage, data)
         nth_img = Meta.parse("gimgArray$(index)")
         for n in 1:size(imgArray, obsdim)
             global g = n
-            preprocess(name*"/$n", convert(PNG, TBImage(eval(nth_img), format)), data)
+            preprocess(name*"/$n", convert(PngImage, TBImage(eval(nth_img), format)), data)
         end
```

Comment:
This code is ... not OK. ikt isn't changed in this PR but please can we not let it stand?

I think it can be replaces with the following, though i am not sure where the 2*obsdim comes from.
I would have thought it was just `1*obsdim`
Check the docs on `selectdim`
```
for n in 1:size(imgArray, obsdim)
    nth_img = selectdim(imgArray, 2*obsim, n)
    preprocess(name*"/$n", convert(PngImage, TBImage(nth_img, format)), data)
end
```

Reply:
Absolutely.
Why `2*obsdim`?

Reply:
never mind. I missed your comment

Reply:
For some reason this fails a lot of tests related to
`log_image(logger, "rand/LN", rand(10, 3) ,LN, step = step)`

Reply:
Oh it shouldn't be `2*obsdim`
the `2*` is because it goes `[:,:,:]` so need to go past the `,`s 
so it should be:

```
for n in 1:size(imgArray, obsdim)
    nth_img = selectdim(imgArray, obsim, n)
    preprocess(name*"/$n", convert(PngImage, TBImage(nth_img, format)), data)
end
```

or something like that,

Reply:
I'm pretty confident it should be `1*obsdim`.
With it, tests pass and looking at the output they make sense.

I think he had 2*obsdim because it was a string, with commas between indices.

Reply:
Hah we made the same comment at the same time